src/main/java/apoc/nodes/Nodes.java

```diff
@@ -136,6 +139,94 @@ public boolean connected(@Name("start") Node start, @Name("start") Node end, @Na
         }
     }
 
+    @Procedure
+    @Description("apoc.nodes.collapse([nodes...],[{properties:'overwrite' or 'discard' or 'combine'}]) yield from, rel, to merge nodes onto first in list")
+    public Stream<VirtualPathResult> collapse(@Name("nodes") List<Node> nodes, @Name(value = "config", defaultValue = "") Map<String, Object> config) {
+        if (nodes == null || nodes.isEmpty()) return Stream.empty();
+        if (nodes.size() == 1) return Stream.of(new VirtualPathResult(nodes.get(0), null, null));
+        Set<Node> nodeSet = new LinkedHashSet<>(nodes);
+        RefactorConfig conf = new RefactorConfig(config);
+        VirtualNode first = createVirtualNode(nodeSet, conf);
+        if (first.getRelationships().iterator().hasNext()) {
+            return StreamSupport.stream(first.getRelationships().spliterator(), false)
+                    .map(relationship -> new VirtualPathResult(relationship.getStartNode(), relationship, relationship.getEndNode()));
+        } else {
+            return Stream.of(new VirtualPathResult(first, null, null));
+        }
+    }
+
+    private VirtualNode createVirtualNode(Set<Node> nodes, RefactorConfig conf) {
+        Create create = new Create();
+        Node first = nodes.iterator().next();
+        List<String> labels = Util.labelStrings(first);
+        if (conf.isCollapsedLabel()) {
+            labels.add("Collapsed");
+        }
+        VirtualNode virtualNode = (VirtualNode) create.vNodeFunction(labels, first.getAllProperties());
+        createVirtualRelationships(nodes, virtualNode, first, conf);
+        nodes.stream().skip(1).forEach(node -> {
+            node.getLabels().forEach(label -> { // Set all Label
+                if (!virtualNode.hasLabel(label)) {
```

Comment:
We should move that into the VirtualNode class. To make sure each label is only added once.